tests/integration/conftest.py

```diff
@@ -0,0 +1,213 @@
+from peewee import SqliteDatabase
+from vorta.store.models import (
+    ArchiveModel,
+    BackupProfileModel,
+    EventLogModel,
+    RepoModel,
+    RepoPassword,
+    SchemaVersion,
+    SettingsModel,
+    SourceFileModel,
+    WifiSettingModel,
+)
+from vorta.views.main_window import MainWindow
+
+models = [
+    RepoModel,
+    RepoPassword,
+    BackupProfileModel,
+    SourceFileModel,
+    SettingsModel,
+    ArchiveModel,
+    WifiSettingModel,
+    EventLogModel,
+    SchemaVersion,
+]
+
+
+def pytest_configure(config):
+    sys._called_from_test = True
+    pytest._wait_defaults = {'timeout': 20000}
+    os.environ['LANG'] = 'en'  # Ensure we test an English UI
+
+
+@pytest.fixture(scope='session')
+def qapp(tmpdir_factory):
+    # DB is required to init QApplication. New DB used for every test.
+    tmp_db = tmpdir_factory.mktemp('Vorta').join('settings.sqlite')
+    mock_db = SqliteDatabase(str(tmp_db))
+    vorta.store.connection.init_db(mock_db)
+
+    # Needs to be disabled before calling VortaApp()
+    if sys.platform == 'darwin':
+        cfg = vorta.store.models.SettingsModel.get(key='check_full_disk_access')
+        cfg.value = False
+        cfg.save()
+
+    from vorta.application import VortaApp
+
+    qapp = VortaApp([])  # Only init QApplication once to avoid segfaults while testing.
+
+    yield qapp
+    mock_db.close()
+    qapp.quit()
+
+
+@pytest.fixture(scope='function', autouse=True)
+def create_test_repo(tmpdir_factory):
+    temp_dir = tmpdir_factory.mktemp('repo')
+    repo_path = str(temp_dir)
+
+    subprocess.run(['borg', 'init', '--encryption=none', repo_path], check=True)
+
+    # create source files dir
+    source_files_dir = os.path.join(temp_dir, 'src')
+    os.mkdir(source_files_dir)
+
+    # /src/file
+    file_path = os.path.join(source_files_dir, 'file')
+    with open(file_path, 'w') as f:
+        f.write('test')
+
+    # /src/dir/
+    dir_path = os.path.join(source_files_dir, 'dir')
+    os.mkdir(dir_path)
+
+    # /src/dir/file
+    file_path = os.path.join(dir_path, 'file')
+    with open(file_path, 'w') as f:
+        f.write('test')
+
+    # Create first archive
+    subprocess.run(['borg', 'create', f'{repo_path}::test-archive1', source_files_dir], cwd=temp_dir, check=True)
+
+    # /src/dir/symlink
+    symlink_path = os.path.join(dir_path, 'symlink')
+    os.symlink(file_path, symlink_path)
+
+    # /src/dir/hardlink
+    hardlink_path = os.path.join(dir_path, 'hardlink')
+    os.link(file_path, hardlink_path)
+
+    # /src/dir/fifo
+    fifo_path = os.path.join(dir_path, 'fifo')
+    os.mkfifo(fifo_path)
+
+    # /src/dir/socket
+    socket_path = os.path.join(dir_path, 'socket')
+    os.mknod(socket_path, mode=0o600 | 0o140000)
+
+    # /src/dir/chrdev
+    chrdev_path = os.path.join(dir_path, 'chrdev')
+    os.mknod(chrdev_path, mode=0o600 | 0o020000)
+
+    # /src/dir/blkdev
+    # blkdev_path = os.path.join(dir_path, 'blkdev')
+    # os.mknod(blkdev_path, mode=0o600 | 0o060000)
```

Comment:
Creating this file type seems to require root-level permission on Linux. Should I skip this?

Reply:
In that case yes. We wouldn't want to run tests with root permissions.